plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -775,6 +775,14 @@ func RunGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercach
 	if !fit {
 		return fit, err
 	}
+	// TODO(HaiyangDING): Make a proper fix for it.
+	allocatable := node.Status.Allocatable
+	allowedPodNumber := allocatable.Pods().Value()
+	if int64(len(nodeInfo.Pods()))+1 > allowedPodNumber {
+		return false,
+			newInsufficientResourceError(podCountResourceName, 1, int64(len(nodeInfo.Pods())), allowedPodNumber)
+	}
+
```

Comment:
What we should do is to insted of calling podFitsResourcesInternal above, call just PodFitsResources and remove it from here.
Then we can even remove podFitsResourcesInternal completely since it will not be used.